src/tests/mock_build/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.5)
+project(mock_build)
+
+set(SOURCE_FILES
+        test_dictionary.h
+        test_dictionary.c
+    )
+
+if(USE_ARDUINO)
+    set(${PROJECT_NAME}_BOARD       ${BOARD})
+    set(${PROJECT_NAME}_PROCESSOR   ${PROCESSOR})
+    set(${PROJECT_NAME}_MANUAL      ${MANUAL})
+    set(${PROJECT_NAME}_PORT        ${PORT})
+    set(${PROJECT_NAME}_SERIAL      ${SERIAL})
+
+    set(${PROJECT_NAME}_SKETCH      dictionary.ino)
+
+    set(${PROJECT_NAME}_SRCS
+        ${SOURCE_FILES}
+        ../../file/kv_stdio_intercept.h
+        ../../file/SD_stdio_c_iface.h
```

Comment:
Remove the `SD_stdio_c_iface` dependency - this is for targeting the device build. The "SD Standard I/O C Interface" is not needed since you're not doing any disk accesses in the code.